core/controllers/reader_test.py

```diff
@@ -115,41 +116,47 @@ def setUp(self):
         self.enable_string_classifier = self.swap(
             feconf, 'ENABLE_STRING_CLASSIFIER', True)
 
-        # Reading YAML exploration into a dictionary.
-        yaml_path = os.path.join(os.path.dirname(os.path.realpath(__file__)),
-                                 '../tests/data/string_classifier_test.yaml')
-        with open(yaml_path, 'r') as yaml_file:
-            self.yaml_content = yaml_file.read()
-
         self.login(self.VIEWER_EMAIL)
         self.signup(self.VIEWER_EMAIL, self.VIEWER_USERNAME)
 
         # Load demo exploration.
-        self.exp_id = '0'
+        self.unused_exp_id = '0'
```

Comment:
Oh, it uses the exp_id as 16 but it is not using the "exp_services.load_demo()" method which loads explorations from files on the basis of their IDs. In fact, it is directly loading the exploration from string_classifier_test.yaml. So, using 16 as the exp_id isn't much of a problem here, but, I agree that it should have been something else.